spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/SparkParquetWriters.java

```diff
@@ -63,8 +64,9 @@ private SparkParquetWriters() {}
 
   @SuppressWarnings("unchecked")
   public static <T> ParquetValueWriter<T> buildWriter(StructType dfSchema, MessageType type) {
+    StructType writeSchema = PruneNullType.prune(dfSchema);
```

Comment:
Is the Iceberg type already pruned, or is unknown ignored when it is missing from the Spark schema?

Reply:
Ah, assuming that the Iceberg schema is coming from `SparkFileWriterFactory`, it should be the already-pruned data schema.